src/tester.rs

```diff
@@ -14,23 +14,34 @@ use std::{
     io::{self, Write},
     path::{Path, PathBuf},
     process::{self, Command},
+    sync::{Arc, Mutex},
 };
 
 use getopts::Options;
+use num_cpus;
 use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+use threadpool::ThreadPool;
 use walkdir::WalkDir;
 
 use crate::{fuzzy, parser::parse_tests};
 
 pub struct LangTester<'a> {
     test_dir: Option<&'a str>,
-    test_file_filter: Option<Box<Fn(&Path) -> bool>>,
-    test_extract: Option<Box<Fn(&str) -> Option<String>>>,
-    test_cmds: Option<Box<Fn(&Path) -> Vec<(&str, Command)>>>,
     use_cmdline_args: bool,
+    test_file_filter: Option<Box<Fn(&Path) -> bool>>,
     cmdline_filters: Option<Vec<String>>,
+    inner: Arc<LangTesterInner>,
 }
 
+struct LangTesterInner {
```

Comment:
It looks like you're splitting this struct so that you can share it between threads using only a single `Arc`. Perhaps `LangTesterShared` or `...SharedInner` (Although that's a bit long-winded) would make it's intention clearer.

Reply:
I agree: the name I currently have sucks. `LangTesterShared` is better, though still not perfect. Suggestions welcome!

Reply:
`LangTesterPoolable`?

Reply:
Yes that's better.